test/model/message_list_test.dart

```diff
@@ -1780,32 +1824,6 @@ void checkInvariants(MessageListView model) {
   check(model.items).length.equals(i);
 }
 
-extension MessageListRecipientHeaderItemChecks on Subject<MessageListRecipientHeaderItem> {
-  Subject<Message> get message => has((x) => x.message, 'message');
-}
-
-extension MessageListDateSeparatorItemChecks on Subject<MessageListDateSeparatorItem> {
-  Subject<Message> get message => has((x) => x.message, 'message');
-}
-
-extension MessageListMessageItemChecks on Subject<MessageListMessageItem> {
-  Subject<Message> get message => has((x) => x.message, 'message');
-  Subject<ZulipContent> get content => has((x) => x.content, 'content');
-  Subject<bool> get showSender => has((x) => x.showSender, 'showSender');
-  Subject<bool> get isLastInBlock => has((x) => x.isLastInBlock, 'isLastInBlock');
-}
-
-extension MessageListViewChecks on Subject<MessageListView> {
```

Comment:
So, the downside of moving these checks-extensions to their own file is that its name `message_list_checks.dart` starts the same way as the test file `message_list_test.dart`. That makes it get in the way of tab-completion, when writing the test file's name for a `git log` or `flutter test` command.

For that reason I've been keeping the extensions within the test file when they're pretty specific to the one file, or to that and one or two others. I think these are best kept here for that reason.